hazelcast/src/main/java/com/hazelcast/sql/impl/exec/sort/MergeSort.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.exec.sort;
+
+import com.hazelcast.sql.impl.row.Row;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.PriorityQueue;
+import java.util.Set;
+
+/**
+ * An utility class to perform merge sort with min-heap.
+ */
+public class MergeSort {
+    /** Constant for unlimited number of rows. */
+    public static final int UNLIMITED = 0;
+
+    /** Sources. */
+    private final MergeSortSource[] sources;
+
+    /** Current items. */
+    private final PriorityQueue<SortKey> heap;
+
+    /** Optional limit on the number of returned results. */
+    private final int limit;
```

Comment:
Do we need `limit` in this PR, provided that `LIMIT/OFFSET` is not supported yet?

Reply:
The leftover, removed in the https://github.com/hazelcast/hazelcast/pull/18013/commits/31db68579318da460575c4a7d1b690f85cbf7ad4